diff --git a/compiler/semtypes.nim b/compiler/semtypes.nim
index b34dc36e3..28de8d982 100644
--- a/compiler/semtypes.nim
+++ b/compiler/semtypes.nim
@@ -1226,6 +1226,11 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
 
     if hasType:
       typ = semParamType(c, a[^2], constraint)
+      if typ.kind == tyVar and typ.sons.len > 0 and typ.sons[0].kind == tyRef:
+        message(
+          c.config, a[^2].info, hintPerformance,
+          "using var modifier for a ref type $1 creates double pointer indirection" % $typ.sons[0]
+        )
       # TODO: Disallow typed/untyped in procs in the compiler/stdlib
       if kind in {skProc, skFunc} and (typ.kind == tyTyped or typ.kind == tyUntyped):
         if not isMagic(getCurrOwner(c)):
